includes/wpml/class-sensei-wpml.php

```diff
@@ -137,4 +148,240 @@ public function set_language_details_when_quiz_created( $quiz_id, $lesson_id ) {
+	}
+
+	/**
+	 * Update lesson course before copied.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @internal
+	 *
+	 * @param mixed  $copied_value Copied value.
+	 * @param int    $post_id_from Post ID from.
+	 * @param int    $post_id_to   Post ID to.
+	 * @param string $meta_key     Meta key.
+	 * @return mixed
+	 */
+	public function update_lesson_course_before_copied( $copied_value, $post_id_from, $post_id_to, $meta_key ) {
+		if ( '_lesson_course' !== $meta_key ) {
+			return $copied_value;
+		}
+
+		if ( empty( $copied_value ) ) {
+			return $copied_value;
+		}
+
+		$course_id = (int) $copied_value;
+
+		$target_language_code = apply_filters(
+			// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+			'wpml_element_language_code',
+			null,
+			array(
+				'element_id'   => $post_id_to,
+				'element_type' => 'lesson',
+			)
+		);
+
+		if ( ! $target_language_code ) {
+			// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+			$target_language_code = apply_filters( 'wpml_current_language', null );
+		}
+
+		// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+		return apply_filters( 'wpml_object_id', $course_id, 'course', false, $target_language_code );
+	}
+
+	/**
+	 * Save lessons fields on course translation created.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @internal
+	 *
+	 * @param int $new_course_id New course ID.
+	 */
+	public function update_lesson_properties_on_course_translation_created( $new_course_id ) {
+		if ( 'course' !== get_post_type( $new_course_id ) ) {
+			return;
+		}
+
+		$details = apply_filters(
+			// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+			'wpml_element_language_details',
+			null,
+			array(
+				'element_id'   => $new_course_id,
+				'element_type' => 'course',
+			)
+		);
+		if ( empty( $details ) ) {
+			return;
+		}
+
+		$details = (array) $details;
+		if ( empty( $details['source_language_code'] ) ) {
+			return;
+		}
+
+		// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+		$master_id = apply_filters( 'wpml_object_id', $new_course_id, 'course', false, $details['source_language_code'] );
+		if ( empty( $master_id ) || $master_id === $new_course_id ) {
+			return;
+		}
+
+		$lesson_ids = Sensei()->course->course_lessons( $master_id, 'any', 'ids' );
+		foreach ( $lesson_ids as $lesson_id ) {
+			if ( ! is_int( $lesson_id ) ) {
+				$lesson_id = (int) $lesson_id;
+			}
+
+			// Create translatons if they don't exist.
+			// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+			$is_translated = apply_filters( 'wpml_element_has_translations', '', $lesson_id, 'lesson' );
+			if ( ! $is_translated ) {
+				// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+				do_action( 'wpml_admin_make_post_duplicates', $lesson_id );
```

Comment:
I imagine it is the action that will duplicate the lessons when translating a course, right? Not sure if I'm doing something wrong, but it didn't create the translated lessons for me when translating a course. I did the following:

1. Create a course with modules and lessons.
2. Publish the lessons.
3. Go to the course listing.
4. Click on "+" to translate it.
5. Translate the strings.
6. Go to the listing and filter the posts by language.
7. Edit the post in the translated language.

Then I see the following (course without modules and lessons):

![Screenshot 2024-02-22 at 10 20 11](https://github.com/Automattic/sensei/assets/876340/bb6577d4-205b-441d-8320-2c0cc9196b71)

And it seems they don't exist at all:

![Screenshot 2024-02-22 at 10 25 11](https://github.com/Automattic/sensei/assets/876340/db779fdd-7ab4-402b-8d28-d8cefb58fd4d)

Here is the course I created to translate:

![Screenshot 2024-02-22 at 10 26 05](https://github.com/Automattic/sensei/assets/876340/377cb55d-ff07-4d99-9126-09b1daddd771)

Reply:
The same happened when translating the lessons first, and then translating the course. I suspect I'm doing some wrong step. 🤔